fail2ban/server/filterpoll.py

```diff
@@ -78,6 +79,16 @@ def _delLogPath(self, path):
 		del self.__file404Cnt[path]
 
 	##
+	# Get a modified log path at once
+	#
+	def getModified(self, modlst):
+		for container in self.getLogPath():
+			filename = container.getFileName()
+			if self.isModified(filename):
+				modlst.append(filename)
+		return modlst
```

Comment:
Thanks for the pointer! IMHO what should be done here is that 
1. wait_for should return two things -- `completed_status, return_value`.  Most of the time you don't care anyways what it returns and just wait for it to complete/timeout, and get a parameter `default` to return if timed out
2. getModified should not get any modlst (I see only a single use of it atm) but create empty list inside first, and extend it/return it; or make `modlst=None` by default and if not None, instead of empty list, use `modlst[:]` copy.  Once again -- this is not even necessity atm, since in the only place it is used, list anyway initialized to empty one outside.
3. outside it would look like
```
			_, modlst = Utils.wait_for(self.getModified, self.sleeptime, [])
			for filename in modlst:
```
so it becomes clear and without any possible side-effect from 'get' method